ui-playground/lib/background.js

```diff
@@ -0,0 +1,114 @@
+    if (tabs.length) {
+      var tab = tabs.reduce((active, item) => {
+        if (aTab && item.id === aTab.id) {
+          active.push(item);
+        }
+        return active;
+      }, []);
+      if (tab.length === 0) {
+        chrome.tabs.update(tabs[0].id, {'active': true});
+        // chrome.pageAction.show(tabs[0].id);
+      } else {
+        // chrome.pageAction.show(tab[0].id);
+      }
+    } else {
+      var url = 'http://chilloutandwatchsomecatgifs.com/';
+      // var url = chrome.extension.getURL('catgifs.html') + '#blah:';
+      chrome.tabs.create({'url': url, 'active': true}, (tab) => {
+        console.log(tab);
+        // chrome.pageAction.show(tab.id);
+      });
+    }
+  });
+};
+
+var count = 0;
+
+chrome.notifications.onClicked.addListener(function (notificationId) {
+  // This doesn't work yet.
+  console.log("Notification Clicked!!!", notificationId);
+  chrome.tabs.query({active: true}, function (aTab) {
+    chillOut(aTab);
+  });
+});
+
+var audio = new Audio();
+audio.onerror = function() {
+  console.log("failed to load coin sound: " + audio.src);
+  audio.src = '';
+};
+audio.src = chrome.extension.getURL('sounds/coin.mp3');
+audio.load();
+
+if (chrome.bookmarks && chrome.bookmarks.onCreated) {
+  chrome.bookmarks.onCreated.addListener(function () {
+    if (audio && audio.src) {
+      if (!audio.paused) {
+        if (audio.currentTime < 0.2) {
+          return;
+        }
+        audio.pause();
+        audio.currentTime = 0;
+      }
+
+      // Sometimes, when playing multiple times, readyState is HAVE_METADATA.
+      if (audio.readyState === 0) {  // HAVE_NOTHING
+        console.log('bad ready state: ' + audio.readyState);
+      } else if (audio.error) {
+        console.log('media error: ' + audio.error);
+      } else {
+        audio.play();
+      }
+    }
+  });
+}
+
+chrome.alarms.onAlarm.addListener(function (alarm) {
+  console.log("Alarm!!!", alarm, count);
+  chrome.browserAction.setBadgeBackgroundColor({color: '#' + Math.floor(Math.random() * 255 * 255 * 255).toString(16)});
+  count = (count + 1) % (Math.floor(Math.random() * 10) + 1);
+  if (alarm.name === 'chillOut' && count === 0) {
+    chrome.notifications.create('chillOut', {
+      type: 'basic',
+      iconUrl: 'icon64.png',
+      title: 'Chill Out! ☕️ ✨ 🚲',
+      message: '…and watch some cat gifs!\n ☕️ ✨ 🚲',
+      contextMessage: 'yo. ☕️ ✨ 🚲',
+      isClickable: true
+    });
+    chrome.tabs.query({active: true}, function (aTab) {
+      chillOut(aTab);
+    });
+  }
+});
+
+chrome.alarms.create('chillOut', {periodInMinutes: 1});
+if (chrome.pageAction) {
+  chrome.pageAction.onClicked.addListener(function (aTab) {
+    chillOut(aTab);
+  });
+
+  chrome.tabs.query({active: true}, tabs => {
+    tabs.forEach(tab => {
+      if (tab && tab.id) {
+        chrome.pageAction.show(tab.id);
+      }
+    });
+  });
+
+  chrome.tabs.onActivated.addListener(activeInfo => {
+    chrome.pageAction.show(activeInfo.tabId);
```

Comment:
It seems like a shame to show the page action for every active tab, as the point of a page action is that, unlike a browser action, it's not always visible. Like this, we might wonder how page action differs from browser action.

(One possibility could be that when the alarm is triggered, the page action is shown for the active tab. Then if it is clicked, it calls chillout.)